frontend/src/components/HistoryItem.tsx

```diff
@@ -3,71 +3,88 @@ import { Col, Row } from 'react-bootstrap';
 import TButton from '@components/TButton';
 import UserItem from '@components/UserItem';
 
-import historyData from '@pages/user/buyer/cart/boughtData.json';
-import goodsData from '@pages/discover/goodsData.json';
+export interface BuyerHistoryItemProps {
+  id: number;
+  shop_name: string;
+  shop_image_url: string;
+  thumbnail_url: string;
+  product_name: string;
+  shipment: number;
+  total_price: number;
+  status: 'paid' | 'shipped' | 'delivered';
+  created_at: string;
+}
 
-const HistoryItem = ({ id, user }: { id: number; user: string }) => {
-  const record = historyData.find((item) => item.recordID === id);
-  const firstItem = goodsData.find((item) => record?.items[0].item_id === item.id);
+export interface SellerHistoryItemProps {
+  id: number;
+  product_name: string;
+  thumbnail_url: string;
+  user_name: string;
+  user_image_url: string;
+  shipment: number;
+  total_price: number;
+  status: 'paid' | 'shipped' | 'delivered';
+  created_at: string;
+}
 
-  let Total = 0;
-  if (record != undefined) {
-    Total = record.items.reduce((accumulator, item) => {
-      const foundItem = goodsData.find((goods) => goods.id === item.item_id);
+interface Props {
+  data: BuyerHistoryItemProps | SellerHistoryItemProps;
+  user: 'buyer' | 'seller';
+}
```

Comment:
```suggestion
export interface BuyerHistoryItemProps {
  user: 'buyer';
  id: number;
  shop_name: string;
  shop_image_url: string;
  thumbnail_url: string;
  product_name: string;
  shipment: number;
  total_price: number;
  status: 'paid' | 'shipped' | 'delivered';
  created_at: string;
}

export interface SellerHistoryItemProps {
  user: 'seller';
  id: number;
  product_name: string;
  thumbnail_url: string;
  user_name: string;
  user_image_url: string;
  shipment: number;
  total_price: number;
  status: 'paid' | 'shipped' | 'delivered';
  created_at: string;
}

interface Props {
  data: BuyerHistoryItemProps | SellerHistoryItemProps;
}
```

you can then use conditional types like such

```ts
if (data.user === 'buyer') {
    // do something with `data.shop_name`
}
```

Reply:
https://www.totaltypescript.com/concepts/mapped-type

Reply:
> https://www.totaltypescript.com/concepts/mapped-type

[this can't be used in interface](https://bobbyhadz.com/blog/typescript-mapped-type-may-not-declare-properties-or-methods) so i use your previous approach